R/utils.R

```diff
@@ -437,18 +442,187 @@ rot_x <- function(angle=30, hjust = 1, vjust = NULL, vertical = FALSE, ...) {
     }
     angle <- 90
     vjust <- 0.5
+  } else {
+    if(is.character(hjust)) {
+      abort("hjust must be numeric or NULL in this case.")
+    }
   }
   x <- element_text(angle = angle, hjust = hjust, vjust = vjust, ...)
   theme(axis.text.x=x)
 }
 
 ##' @rdname rot_x
 ##' @export
-rot_y <- function(angle=30, hjust = 1, vjust = NULL,...) {
-  y <- element_text(angle = angle, hjust = hjust, vjust = vjust, ...)
+rot_y <- function(angle=30, hjust = 1, vjust = NULL, vertical = FALSE, ...) {
+  if(isTRUE(vertical)) {
+    if(is.character(vjust)) {
+      vjust <- match.arg(vjust, c("left", "right"))
+      if(vjust=="left") vjust <- 1
+      if(vjust=="right") vjust <- 0
```

Comment:
Hmm, this would get into trouble with partial matches.

```R
match.arg("l", c("left", "right"))
#> [1] "left"
```

Convert to a `switch` statement or to `rlang::arg_match` (which doesn't allow partial matches)?

Reply:
Switch to `rlang::arg_match()`.